### PR TITLE
Create the Main Structure for Admin Register Application Flow

### PR DESCRIPTION
**Pull request by AhmetAksunger**
### Description

Created the main classes for the admin register application flow. The main reason for this pull request to be created is to avoid possible conflicts while developing the register application flow in parallel with other developers.

### Related Issue

There's no direct related issue with this pull request. 

### Proposed Changes

Please provide a brief and descriptive title for your pull request, summarizing the changes or features introduced. It
should be in English and clearly indicate the purpose of the pull request. Optionally, you can provide a more detailed
description of what you have done.

- Branches should follow the below `Naming Conventions of Branchs` header in [README.md](https://github.com/afet-yonetim-sistemi/ays-be/edit/main/README.md) File

### Additional Information

If there's any additional information or context that the reviewers should know, please provide it here.

### Checklist

Please check the following before submitting your pull request:

- [x] I have tested my changes locally and they are working as expected.
- [x] The code is formatted according to the project's coding guidelines and style.
- [x] Necessary documentation has been added or existing documentation has been updated.
- [x] Relevant unit tests have been written and included.
- [x] Relevant integration tests have been written and included.
- [x] The code has been reviewed to ensure its quality.
- [x] The code does not contain any issues flagged by SonarLint.
- [x] Default reviewers have been assigned to this pull request.
- [x] Assignees have been added if necessary.
- [x] Labels have been applied if necessary.
- [x] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

### Reviewers

The default project maintainers and team members will be automatically assigned as reviewers for this pull request.

### CODE REVIEW

**Review by agitrubard — CHANGES_REQUESTED** [review]
Oluşturulan sınıflarda herhangi bir problem yok, eline sağlık 🙌🏼 

Ortak kullanılacak başka sınıflar var mı? Entity olabilir, domain model olabilir

**Comment by AhmetAksunger** [comment]
> Oluşturulan sınıflarda herhangi bir problem yok, eline sağlık 🙌🏼
> 
> Ortak kullanılacak başka sınıflar var mı? Entity olabilir, domain model olabilir

Entity zaten oluşturulmuş. Ekstradan domain modeli ve entity -> domain model mapper sınıflarını oluşturdum.

**Review by agitrubard — DISMISSED** [review]

**Review by ibrahimdenizz — APPROVED** [review]
LGTM!

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterApplicationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.ays.admin_user.service.impl;
+
+import com.ays.AbstractUnitTest;
+
+public class AdminUserRegisterApplicationServiceImplTest extends AbstractUnitTest {
```
Comment (ibrahimdenizz) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterApplicationServiceImplTest.java:5]: Mock repository ve servisi ekliyebiliriz

File: src/test/java/com/ays/admin_user/controller/AdminUserRegisterApplicationControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.ays.admin_user.controller;
+
+import com.ays.AbstractRestControllerTest;
+
+public class AdminUserRegisterApplicationControllerTest extends AbstractRestControllerTest {
```
Comment (ibrahimdenizz) [inline src/test/java/com/ays/admin_user/controller/AdminUserRegisterApplicationControllerTest.java:5]: Mock servisi ekliyebiliriz.

**Review by ibrahimdenizz — CHANGES_REQUESTED** [review]

**Comment by agitrubard** [comment]
@AhmetAksunger Selamlar,
@ibrahimdenizz'in yazdığı yorumlara bakabilir misin? Bu geliştirmeyi mergeleyelim, diğer geliştirmelerin önü açılsın

**Review by anilhazar — APPROVED** [review]

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterApplicationServiceImplTest.java (hunk 1 above)
Comment (AhmetAksunger) [reply src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterApplicationServiceImplTest.java]: Ekledim.

File: src/test/java/com/ays/admin_user/controller/AdminUserRegisterApplicationControllerTest.java (hunk 2 above)
Comment (AhmetAksunger) [reply src/test/java/com/ays/admin_user/controller/AdminUserRegisterApplicationControllerTest.java]: Ekledim.

**Review by ibrahimdenizz — APPROVED** [review]

**Review by agitrubard — APPROVED** [review]

**Pull request merged**